x/metadata/keeper/mocks_test.go

```diff
@@ -325,14 +332,323 @@
+	t.Helper()
+	act := k.Calls.MintCoins
+	if assert.Equal(t, exp, act, "Calls made to MintCoins") {
+		return true
+	}
+	expStrs := mapToStrings(exp)
+	actStrs := mapToStrings(act)
+	assert.Equal(t, expStrs, actStrs, "Calls (as strings) made to MintCoins")
+	return false
+}
+
+// AssertBurnCoinsCalls asserts that calls made to BurnCoins are as expected.
+func (k *MockBankKeeper) AssertBurnCoinsCalls(t *testing.T, exp []*MintBurnCall) bool {
+	t.Helper()
+	act := k.Calls.BurnCoins
+	if assert.Equal(t, exp, act, "Calls made to BurnCoins") {
+		return true
+	}
+	expStrs := mapToStrings(exp)
+	actStrs := mapToStrings(act)
+	assert.Equal(t, expStrs, actStrs, "Calls (as strings) made to BurnCoins")
+	return false
+}
+
+// AssertSendCoinsCalls asserts that calls made to SendCoins are as expected.
+func (k *MockBankKeeper) AssertSendCoinsCalls(t *testing.T, exp []*SendCoinsCall) bool {
+	t.Helper()
+	act := k.Calls.SendCoins
+	if assert.Equal(t, exp, act, "Calls made to SendCoins") {
+		return true
+	}
+	expStrs := mapToStrings(exp)
+	actStrs := mapToStrings(act)
+	assert.Equal(t, expStrs, actStrs, "Calls (as strings) made to SendCoins")
+	return false
+}
+
+// AssertDenomOwnerCalls asserts that calls made to DenomOwner are as expected.
+func (k *MockBankKeeper) AssertDenomOwnerCalls(t *testing.T, exp []string) bool {
+	t.Helper()
+	return assert.Equal(t, exp, k.Calls.DenomOwner, "Calls made to DenomOwner")
+}
+
+// MintBurnCall is the args provided to either MintCoins or BurnCoins.
+type MintBurnCall struct {
+	ModuleName string
+	Coins      sdk.Coins
+}
+
+func (c MintBurnCall) String() string {
+	return c.ModuleName + "_" + c.Coins.String()
+}
+
+func NewMintBurnCall(moduleName string, coins sdk.Coins) *MintBurnCall {
+	return &MintBurnCall{
+		ModuleName: moduleName,
+		Coins:      coins,
+	}
+}
+
+// SendCoinsCall is the args provided to SendCoins.
+type SendCoinsCall struct {
+	FromAddr sdk.AccAddress
+	ToAddr   sdk.AccAddress
+	Amt      sdk.Coins
+}
+
+func (c SendCoinsCall) String() string {
+	return string(c.FromAddr) + "->" + string(c.ToAddr) + ":" + c.Amt.String()
+}
+
+func NewSendCoinsCall(fromAddr, toAddr sdk.AccAddress, amt sdk.Coins) *SendCoinsCall {
+	return &SendCoinsCall{
+		FromAddr: fromAddr,
+		ToAddr:   toAddr,
+		Amt:      amt,
+	}
+}
+
+// DenomOwnerResult is the args returned by DenomOwner.
+type DenomOwnerResult struct {
+	Owner sdk.AccAddress
+	Err   string
+}
+
+func (k *MockBankKeeper) BlockedAddr(addr sdk.AccAddress) bool {
+	k.Calls.BlockedAddr = append(k.Calls.BlockedAddr, addr)
+	return k.BlockedAddrResults[string(addr)]
+}
+
+func (k *MockBankKeeper) MintCoins(_ context.Context, moduleName string, amt sdk.Coins) error {
+	k.Calls.MintCoins = append(k.Calls.MintCoins, NewMintBurnCall(moduleName, amt))
+	if len(k.MintCoinsResults) > 0 {
+		err := k.MintCoinsResults[0]
+		k.MintCoinsResults = k.MintCoinsResults[1:]
+		if len(err) > 0 {
+			return errors.New(err)
+		}
+	}
+	return nil
+}
```

Comment:
There's already perfect control over which calls should return an error, and perfect clarity of what will happen next. `MintCoins` defaults to returning `nil` when `MintCoinsResults` is empty. It also returns `nil` if `MintCoinsResults[0]` is an empty string. Entries are used once then discarded, so `MintCoinsResults` only has info on what is to come. This stuff is all documented in the comments on the methods used to set up these mock keepers. If there's an entry left over in it, either it doesn't matter or the test case is failing because an error isn't as expected.

Using a `map` and counter is more complex than just using a slice since it requires two fields. You also can't look at just the contents of the `map` to determine what will happen next and it's harder to determine if there were entries that weren't used. A slice also has all the functionality needed here, so using some other data structure (e.g. a `Queue` of some sort) would just complicate things without any benefit.